drivers/mil_passive_sonar/scripts/multilateration

```diff
@@ -57,25 +67,82 @@ class MultilaterationNode(object):
         except tf2_ros.TransformException, e:
             rospy.logwarn('TF Exception: {}'.format(e))
             return
+
+        # Check if two samples were taken too close to each other
+        if self.last_pos is not None and np.linalg.norm(self.last_pos - origin) < self.minimum_delta_dist:
+            rospy.loginfo("Observation too close to previous, ignoring")
+            return
+        self.last_pos = origin
+
+        # Add ping to samples list
         self.vec_samples.append((origin, origin + vec))
-        if len(self.vec_samples) == self.vec_samples.maxlen:
+
+        # If there are enough samples, calculate position
+        if len(self.vec_samples) >= self.buffer_min:
             self.publish_position()
 
     def publish_position(self):
-        line_array = np.array([(p[0][0], p[0][1], p[1][0], p[1][1]) for p in self.vec_samples])
+        # Setup lines from samples
+        line_array = np.array([(np.array(p[0][0:2]), np.array(p[1][0:2])) for p in self.vec_samples])
+
+        # Calculate least squares intersection
         where = self.ls_line_intersection2d(line_array)
+
+        # If no point could be determined, don't publish
+        if where is None:
+            return
+
+        # Publish point
         ps = PointStamped()
+        ps.header.frame_id = self.global_frame
         ps.header.stamp = rospy.Time.now()
         ps.point = numpy_to_point(where)
         self.position_pub.publish(ps)
 
-    def ls_line_intersection2d(self, line_array):
+    def ls_line_intersection2d(self, lines_array):
         """
         Find the intersection of lines in the least-squares sense.
-        start - Nx3 numpy array of start points
-        end - Nx3 numpy array of end points
-        https://en.wikipedia.org/wiki/Line-line_intersection#In_two_dimensions_2
+        Filter lobs such that no lob can have more than a number of intersections with other lines within a radius of the boat
         """
+
+        line_array = np.array([[0, 0, 0, 0]])
+        line_intersecting_array = np.array([])
+
+        for line1 in lines_array:
```

Comment:
Comment noting that this section filters out nearby intersections

Reply:
Done